File: Simulator-CSharp/Views/AmountControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ATM.Views
{
    public partial class AmountControl : UserControl
    {
        public AmountControl()
        {
            InitializeComponent();
        }

        private int _Value = 0;
        public int Value
        {
            get { return _Value; }
            set
            {
                if (_Value != value)
                {
                    _Value = value;
                    OnValueChanged(EventArgs.Empty);
                }
            }
        }

        private int _Minimum = 0;
        public int Minimum
        {
            get { return _Minimum; }
            set
            {
                if (_Minimum != value)
                {
                    _Minimum = value;
                    OnMinimumChanged(EventArgs.Empty);
                }
            }
        }
        private int _Maximum = 100;
        public int Maximum
        {
            get { return _Maximum; }
            set
            {
                if (_Maximum != value)
                {
                    _Maximum = value;
                    OnMaximumChanged(EventArgs.Empty);
                }
            }
        }

        #region Events

        private void OnMinimumChanged(EventArgs e)
        {
            numericUpDownAmount.Minimum = this.Minimum;
            trackBarAmount.Minimum = this.Minimum;
        }

        private void OnMaximumChanged(EventArgs e)
        {
            numericUpDownAmount.Maximum = this.Maximum;
            trackBarAmount.Maximum = this.Maximum;
        }

        private void OnValueChanged(EventArgs e)
        {
            numericUpDownAmount.Value = this.Value;
            trackBarAmount.Value = this.Value;
            if (ValueChanged != null)
                ValueChanged(this, EventArgs.Empty);
        }

        public event EventHandler ValueChanged;

        #endregion

        public double Percent()
        {
            return Value * 100 / Maximum;
        }

        private void trackBarAmount_Scroll(object sender, EventArgs e)
        {
            Value = ((TrackBar)sender).Value;
            numericUpDownAmount.Value = Value;
        }

        private void numericUpDownAmount_ValueChanged(object sender, EventArgs e)
        {
            Value = (int)((NumericUpDown)sender).Value;
            trackBarAmount.Value = Value;
        }
    }
}

File: Simulator-VB/Views/AmountControl.vb
Public Class AmountControl

    Private _Value As Integer = 0
    Public Property Value As Integer
        Get
            Return _Value
        End Get
        Set(value As Integer)
            If _Value <> value Then
                _Value = value
                OnValueChanged(EventArgs.Empty)
            End If
        End Set
    End Property

    Private _Minimum As Integer = 0
    Public Property Minimum As Integer
        Get
            Return _Minimum
        End Get
        Set(value As Integer)
            If _Minimum <> value Then
                _Minimum = value
                OnMinimumChanged(EventArgs.Empty)
            End If
        End Set
    End Property
    Private _Maximum As Integer = 100
    Public Property Maximum As Integer
        Get
            Return _Maximum
        End Get
        Set(value As Integer)
            If _Maximum <> value Then
                _Maximum = value
                OnMaximumChanged(EventArgs.Empty)
            End If
        End Set
    End Property

#Region "Events"

    Private Sub OnMinimumChanged(e As EventArgs)
        Me.NumericUpDownAmount.Minimum = Me.Minimum
        Me.TrackBarAmount.Minimum = Me.Minimum
    End Sub

    Private Sub OnMaximumChanged(e As EventArgs)
        Me.NumericUpDownAmount.Maximum = Me.Maximum
        Me.TrackBarAmount.Maximum = Me.Maximum
    End Sub

    Private Sub OnValueChanged(e As EventArgs)
        Me.NumericUpDownAmount.Value = Me.Value
        Me.TrackBarAmount.Value = Me.Value
        RaiseEvent ValueChanged(Me, EventArgs.Empty)
    End Sub

    Public Event ValueChanged As EventHandler

#End Region

    Public Function Percent() As Double
        Return Me.Value * 100 / Me.Maximum
    End Function

    Private Sub TrackBarAmount_Scroll(sender As Object, e As EventArgs) Handles TrackBarAmount.Scroll
        Me.Value = sender.Value
        Me.NumericUpDownAmount.Value = Me.Value
    End Sub

    Private Sub NumericUpDownAmount_ValueChanged(sender As Object, e As EventArgs) Handles NumericUpDownAmount.ValueChanged
        Me.Value = sender.Value
        Me.TrackBarAmount.Value = Me.Value
    End Sub
End Class
